benches/api_calls.rs

```diff
@@ -0,0 +1,225 @@
+        let s = env.new_string("").unwrap();
+        let obj = black_box(JObject::from(s));
+        let method_id = env.get_method_id(
+            obj, METHOD_OBJECT_HASH_CODE, SIG_OBJECT_HASH_CODE).unwrap();
+
+        b.iter(|| jni_call_unsafe(&env, obj, method_id));
+    }
+
+    #[bench]
+    fn jni_new_object_str(b: &mut Bencher) {
+        let env = VM.attach_current_thread().unwrap();
+        let class = CLASS_OBJECT;
+
+        b.iter(|| {
+            let obj = env.new_object(class, SIG_OBJECT_CTOR, &[]).unwrap();
+            env.delete_local_ref(obj).unwrap();
+        });
+    }
+
+    #[bench]
+    fn jni_new_object_by_id_str(b: &mut Bencher) {
+        let env = VM.attach_current_thread().unwrap();
+        let class = CLASS_OBJECT;
+        let ctor_id = env.get_method_id(class, METHOD_CTOR, SIG_OBJECT_CTOR).unwrap();
+
+        b.iter(|| {
+            let obj = env.new_object_by_id(class, ctor_id, &[]).unwrap();
+            env.delete_local_ref(obj).unwrap();
+        });
+    }
+
+    #[bench]
+    fn jni_new_object_jclass(b: &mut Bencher) {
+        let env = VM.attach_current_thread().unwrap();
+        let class: JClass = CLASS_OBJECT.lookup(&env).unwrap();
+
+        b.iter(|| {
+            let obj = env.new_object(class, SIG_OBJECT_CTOR, &[]).unwrap();
+            env.delete_local_ref(obj).unwrap();
+        });
+    }
+
+    #[bench]
+    fn jni_new_object_by_id_jclass(b: &mut Bencher) {
+        let env = VM.attach_current_thread().unwrap();
+        let class: JClass = CLASS_OBJECT.lookup(&env).unwrap();
+        let ctor_id = env.get_method_id(class, METHOD_CTOR, SIG_OBJECT_CTOR).unwrap();
+
+        b.iter(|| {
+            let obj = env.new_object_by_id(class, ctor_id, &[]).unwrap();
+            env.delete_local_ref(obj).unwrap();
+        });
+    }
+
+    #[bench]
+    fn jni_new_global_ref(b: &mut Bencher) {
+        let env = VM.attach_current_thread().unwrap();
+        let class = CLASS_OBJECT;
+        let obj = env.new_object(class, SIG_OBJECT_CTOR, &[]).unwrap();
+        let global_ref = env.new_global_ref(obj).unwrap();
+        env.delete_local_ref(obj).unwrap();
+
+        b.iter(|| env.new_global_ref(global_ref.as_obj()).unwrap());
+    }
+
+    #[bench]
+    fn jni_get_java_vm(b: &mut Bencher) {
+        let env = VM.attach_current_thread().unwrap();
+
+        b.iter(|| {
+            let _jvm = env.get_java_vm().unwrap();
+        });
+    }
+
+    #[bench]
+    fn native_arc(b: &mut Bencher) {
+        let env = VM.attach_current_thread().unwrap();
+        let class = CLASS_OBJECT;
+        let obj = env.new_object(class, SIG_OBJECT_CTOR, &[]).unwrap();
+        let global_ref = env.new_global_ref(obj).unwrap();
+        env.delete_local_ref(obj).unwrap();
+        let arc = Arc::new(global_ref);
+
+        b.iter(|| {
+            let _ = black_box(Arc::clone(&arc));
+        });
+    }
+
+    #[bench]
+    fn native_rc(b: &mut Bencher) {
```

Comment:
Why are there Arc/Rc tests?

Reply:
They are needed for comparison when choosing particular solutions.

Reply:
Which solutions? They don't seem to be a part of the jni-rs API, and therefore, it's unclear how users of jni-rs will benefit from them.

Reply:
For example, when designing `GlobalRef`, I preferred to use `Arc`, rather than creating a new reference using `JNIEnv::new_global_ref()`. Users of jni-rs can have similar situations and it will be useful for them to compare the cost of function calls with JNI and some native tools.